### PR TITLE
Deprecate users/{userId}/deliveryServices

### LINKED ISSUE #4402: Provide the ability to fetch all delivery services accessible to a certain tenant
## I'm submitting a ...

- [ ] bug report
- [x] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:
There is no current way to pull all delivery services accessible to a given tenant. I.e. imagine the following tenant tree:

```
- root
-- tenant_1
--- tenant_1.1
```

Currently, you can fetch all delivery services assigned to a tenant as such:

GET /deliveryservices?tenant=tenant_1

but you cannot fetch all delivery services "accessible to" tenant_1 which means all delivery services assigned to tenant_1 AND tenant_1.1

## Expected / new behavior:
Provide a query param to GET /deliveryservices that allows the user to fetch all delivery services accessible to a certain tenant. i.e. GET /deliveryservices?accessibleTo=tenant_1 which will provide all the delivery services assigned to tenant_1 OR tenant_1.1

## Anything else:
Once this is done, the following API route can be deprecated in 2.x:

GET users/{id}/deliveryservices

### PR DESCRIPTION
**Pull request by shamrickus**
## What does this PR (Pull Request) do?

- [X] This PR fixes #4402, deprecating `users/{id}/deliveryservices` and adding the `accessibleTo` query parameter to `deliveryservices` 

## Which Traffic Control components are affected by this PR?

- Traffic Control Client 
- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?
Ensure docs build, TO tests pass, TP tests pass.

For TO:
- Verify `1.1/users/:id/deliveryservices` returns a deprecation notice (This endpoint does not work see #3754)
- Verify `2.0/users/:id/deliveryservices` doesn't exist
- Verify `2.0/deliveryservices?accessibleTo={tenantId}` works and returns all delivery services that have this tenant, or any delivery services that are children of the given tenant.

For TP:
- Verify when clicking the `View Delivery Services` button on the user editting screen (`users/{userId}`), it takes the user to `tenant/{tenantId}/deliveryservices`
- Verify that `tenant/{tenantId}/deliveryservices` returns the delivery services referencing this tenant, **and** this tenants children (previously does not show children).

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_portal/app/src/common/modules/form/user/FormUserController.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ var FormUserController = function(user, $scope, $location, formUtils, stringUtil
     $scope.labelize = stringUtils.labelize;
 
     $scope.viewDeliveryServices = function() {
-        $location.path($location.path() + '/delivery-services');
+        $location.path('/tenants/' + userModel.user.tenantId + '/delivery-services');
```
Comment (mitchell852) [inline traffic_portal/app/src/common/modules/form/user/FormUserController.js]: fyi for whoever reviews this. i was the one that suggested when viewing a user's delivery services, it's probably easiest to simply send them to view the delivery services of the user's tenant instead.

this does mean, however, that we are abandoning a view (users/id/deliveryservices) that should probably be cleaned up.

Comment (mitchell852) [reply traffic_portal/app/src/common/modules/form/user/FormUserController.js]: Here is a PR where @ocket8888 removed a view. In this case he removed the ability to view cachegroups for a parameter - https://github.com/apache/trafficcontrol/pull/3636

File: traffic_ops/testing/api/v2/deliveryservices_test.go [hunk 2]
```diff
@@ -296,6 +298,53 @@ func UpdateDeliveryServiceWithInvalidSliceRangeRequest(t *testing.T) {
...
+	tenant = &resp.Response
+
+	//No delivery services are associated with this new tenant
+	err = getByTenants(tenant.ID, 0)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	//First and only child tenant, no access to root
+	err = getByTenants(3, len(testData.DeliveryServices) - 1)
```
[the remark this answers is not in the document]
Comment (shamrickus) [reply traffic_ops/testing/api/v2/deliveryservices_test.go]: We will, the test failures are from that. I was only running the individual tests so the id was always the same, but we can't rely on execution order (and therefore id)

File: traffic_portal/app/src/common/modules/form/user/FormUserController.js (hunk 1 above)
Comment (mitchell852) [inline traffic_portal/app/src/common/modules/form/user/FormUserController.js]: So what this does is it directs you to the logged in user's tenant delivery services page every time which is not what you want. You want this to direct you to the user that was clicked's delivery services page. 

For example, if i'm on the this page: https://tp.domain.com/#!/users/244, then clicking "view delivery services" should direct me to `$location.path('/tenants/' + user[244].tenantId + '/delivery-services');`

Comment (mitchell852) [reply traffic_portal/app/src/common/modules/form/user/FormUserController.js]: actually, i think you an just get rid of the `userModel` part.....or you can pass user into the viewDeliveryServices function if you want to be more explicit.

**Pull request merged**